lib/matplotlib/path.py

```diff
@@ -620,11 +623,20 @@ def get_extents(self, transform=None, **kwargs):
             self = transform.transform_path(self)
         if self.codes is None:
             xys = self.vertices
+        # Short cut the case with only straight lines
         elif len(np.intersect1d(self.codes, [Path.CURVE3, Path.CURVE4])) == 0:
-            xys = self.vertices[self.codes != Path.CLOSEPOLY]
+            draw_idxs = np.in1d(self.codes, Path.LINETO)
+            # Include the start and end point of each line
+            draw_idxs = draw_idxs | np.roll(draw_idxs, -1)
+            # Add in commands that aren't ignored
+            # (in the straight line case here this is only MOVETO)
+            draw_idxs = draw_idxs | ~np.in1d(self.codes, Path.IGNORED_CODES)
+            xys = self.vertices[draw_idxs]
```

Comment:
IMHO this way of writing is more concise:

```suggestion
            # we have only straight lines:
            # consider line starts, line ends and points moved to
            line_end_mask = np.isin(self.codes, Path.LINETO)
            line_start_mask = np.roll(line_end_mask, -1)
            moveto_mask = np.isin(self.codes, Path.MOVETO)
            xys = self.vertices[line_start_mask | line_end_mask | moveto_mask]
```

Note also that I've used `np.isin()` instead of `np.in1d()` because `self.codes` are 1D.

*Edit*: I'm wondering if 
~~~
xys = self.vertices[np.isin(self.codes, [Path.LINETO, Path.MOVETO])]
~~~
would be sufficient. Here, the codes of points at line starts cannot be STOP, CURVE3, CURVE4. The only grey area would be CLOSEPOLY followed by LINETO, in which case the line start would be a CLOSEPOLY node. IMHO that's not quite well defined CLOSEPOLY should always be followed by MOVETO.

Reply:
I think they can be CURVE{3,4}, as there will always be a final control point for those curves (the curve endpoint) which one can then draw a straight line from.

Reply:
But the `elif` conditions explicitly excludes CURVE{3,4} here.

Reply:
Sorry for the lag. A couple of things: 

> The only grey area would be CLOSEPOLY followed by LINETO, in which case the line start would be a CLOSEPOLY node. IMHO that's not quite well defined CLOSEPOLY should always be followed by MOVETO.

While "undefined" is a reasonable guess, AFAICT, all backends currently handle this behavior consistently. That is, CLOSEPOLY's vertex is ignored and it is treated as "(start_point, LINETO)" plus adding the extra cap. 

Which means that CLOSEPOLY followed by LINETO is well-defined, and does the natural thing:
```python
import matplotlib.pyplot as plt
from matplotlib.path import Path
from matplotlib.patches import PathPatch

p = Path(
    [[0, 0], [0, 1], [1, 1], [5, 5], [1, 0], [0, -1], [-5, 5]], 
    [Path.MOVETO, Path.LINETO, Path.LINETO, Path.CLOSEPOLY, 
     Path.LINETO, Path.LINETO, Path.CLOSEPOLY]
)
fig, ax = plt.subplots()
ax.add_patch(PathPatch(p))
ax.set_xlim([-0.5, 1.5])
ax.set_ylim([-1.5, 1.5])
```
![image](https://user-images.githubusercontent.com/1475390/103596556-0d0cfe80-4eb3-11eb-9f31-0cc2761db016.png)

And, unless I'm missing something, then 


> I'm wondering if
> 
> ```
> xys = self.vertices[np.isin(self.codes, [Path.LINETO, Path.MOVETO])]
> ```
> 
> would be sufficient. 

tl;dr: yes. We should just be doing this.